src/engine/MILPEncoder.cpp

```diff
@@ -126,3 +129,85 @@ void MILPEncoder::encodeReLUConstraint( GurobiWrapper &gurobi, ReluConstraint
     terms.append( GurobiWrapper::Term( -sourceUb, Stringf( "a%u", index ) ) );
     gurobi.addLeqConstraint( terms, 0 );
 }
+
+void MILPEncoder::encodeMaxConstraint( GurobiWrapper &gurobi, MaxConstraint *max )
+{
+
+    if ( !max->isActive() )
+        return;
+
+    // y = max(x_1, x_2, ... , x_m)
+    unsigned y = max->getF();
+
+    // xs = [x_1, x_2, ... , x_m]
+    List<unsigned> xs = max->getParticipatingVariables();
+    unsigned m = xs.size();
+
+    // upper bounds of each x_i
+    double* ubs = new double[m];
+
+    // terms for Gurobi
+    List<GurobiWrapper::Term> terms;
+
+    unsigned i = 0;
+    for ( const auto &x : xs ) {
+        // add binary variable
+        gurobi.addVariable( Stringf( "a%u", x ),
+                            0,
+                            1,
+                            GurobiWrapper::BINARY );
+
+        terms.append( GurobiWrapper::Term( 1, Stringf( "a%u", x ) ) );
+        ubs[i] = _tableau.getUpperBound( x );
+        i++;
+    }
+
+    // add constraint: a_1 + a_2 + ... + a_m = 1
+    gurobi.addEqConstraint( terms, 1 );
+
+    terms.clear();
+
+    i = 0;
+    for ( const auto &x : xs ) {
+        // add constraint: y <= x_i + (1 - a_i) * (umax - l)
+        double umax = getUmax( ubs, i, m );
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", y ) ) );
+        terms.append( GurobiWrapper::Term( -1, Stringf( "x%u", x ) ) );
+        terms.append( GurobiWrapper::Term( umax - _tableau.getLowerBound( x ), Stringf( "a%u", x ) ) );
+        gurobi.addLeqConstraint( terms, umax - _tableau.getLowerBound( x ) );
+
+        terms.clear();
+
+        // add constraint: y >= x_i
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", y ) ) );
```

Comment:
"y >= x_i" is already encoded as an inequality here: https://github.com/NeuralNetworkVerification/Marabou/blob/3bf23beb8a9ce3d2724bf63984af792ec006c370/src/engine/MaxConstraint.cpp#L489-L515.
I think you could just remove line 182 - 188.

Reply:
will do.

Reply:
done.